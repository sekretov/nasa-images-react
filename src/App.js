import React from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import './App.css';
import NasaImage from './NasaImage.js';
import NasaImageList from './NasaImageList.js';


class App extends React.Component {
    constructor() {
        super();

        this.state = {
            'images': null
        };
    }

    componentDidMount() {
        fetch('https://images-api.nasa.gov/search?q=apollo%2011&description=moon%20landing&media_type=image')
            .then(response => response.json())
            .then(data => {
                this.setState({
                    'images': data.collection.items
                });
            });
    }

    render() {
        return (
            <main className="container-fluid">
                <header className="row">
                    <div className="col"><h2>Nasa Images</h2></div>
                </header>
                <div className="row">
                    <Router>
                        <nav className="col-4">
                            <NasaImageList images = {this.state.images} />
                        </nav>
                        <article className="col">
                            <Route path="/image/:nasa_id" component={NasaImage} />
                        </article>
                    </Router>
                </div>
            </main>
        );
    }
}

export default App;
