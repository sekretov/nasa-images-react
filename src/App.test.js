import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without children without crashing', () => {
  const wrapper = shallow(<App />);
  const title = <h1>Nasa Images</h1>;

  expect(wrapper.contains(title)).toEqual(false); //содержит

});

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});