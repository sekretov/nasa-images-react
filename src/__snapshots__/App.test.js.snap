// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="container-fluid"
>
  <header
    className="row"
  >
    <div
      className="col"
    >
      <h2>
        Nasa Images
      </h2>
    </div>
  </header>
  <div
    className="row"
  >
    <nav
      className="col-4"
    >
      <ul>
        Loading...
      </ul>
    </nav>
    <article
      className="col"
    />
  </div>
</main>
`;
